ncs/encrypt_script.py

```diff
@@ -0,0 +1,329 @@
+#
+# Copyright (c) 2024 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+"""
+This script allows to output artifacts needed by a SUIT envelope for encrypted firmware.
+"""
+
+import os
+import cbor2
+from argparse import ArgumentParser
+from argparse import RawTextHelpFormatter
+from pathlib import Path
+from cryptography.hazmat.primitives import hashes
+from cryptography.hazmat.backends import default_backend
+from enum import Enum, unique
+from pynrfkms.kms import KMS
+import getpass
+
+
+@unique
+class SuitAlgorithms(Enum):
+    """Suit algorithms."""
+
+    COSE_ALG_AES_GCM_128 = 1
+    COSE_ALG_AES_GCM_192 = 2
+    COSE_ALG_AES_GCM_256 = 3
+    COSE_ALG_A128KW = -3
+    COSE_ALG_A192KW = -4
+    COSE_ALG_A256KW = -5
+
+
+class SuitIds(Enum):
+    """Suit elements identifiers."""
+
+    COSE_ALG = 1
+    COSE_KEY_ID = 4
+    COSE_IV = 5
+
+
+class SuitDomains(Enum):
+    """Suit domains."""
+
+    APPLICATION = "application"
+    RADIO = "radio"
+    CELL = "cell"
+    WIFI = "wifi"
+
+    def __str__(self):
+        return self.value
+
+
+class SuitDigestAlgorithms(Enum):
+    """Suit digest algorithms."""
+
+    SHA_256 = "sha-256"
+    SHA_384 = "sha-384"
+    SHA_512 = "sha-512"
+    SHAKE128 = "shake128"
+    SHAKE256 = "shake256"
+
+    def __str__(self):
+        return self.value
+
+
+class EncryptionKMSBackends(Enum):
+    """kms backends."""
+
+    VAULT = "vault"
+    LOCAL = "local"
+
+    def __str__(self):
+        return self.value
+
+
+KEY_IDS = {
+    SuitDomains.APPLICATION.value: 0x40020200,
+    SuitDomains.RADIO.value: 0x40030200,
+    SuitDomains.CELL.value: 0x40040200,
+    SuitDomains.WIFI.value: 0x40060200,
+}
+
+
+class DigestGenerator:
+
+    _hash_func = {
+        SuitDigestAlgorithms.SHA_256.value: hashes.SHA256(),
+        SuitDigestAlgorithms.SHAKE128.value: hashes.SHAKE128(16),
+        SuitDigestAlgorithms.SHA_384.value: hashes.SHA384(),
+        SuitDigestAlgorithms.SHA_512.value: hashes.SHA512(),
+        SuitDigestAlgorithms.SHAKE256.value: hashes.SHAKE256(32),
+    }
+
+    def __init__(self, hash_name: str):
+        """Initialize object."""
+        if hash_name not in self._hash_func.keys():
+            raise ValueError(f"Unsupported hash algorithm: {hash_name}")
+        self._hash_name = hash_name
+
+    def generate_digest_for_plain_text(self, plaintext_file_path: Path, output_directory: Path):
```

Comment:
Since this method generates two files - I'd include the `size` in the method name too.

Reply:
Method name changed.